### PR TITLE
Eclass whitelist update

### PR DESCRIPTION
**Pull request by SoapGentoo**

### CODE REVIEW

File: eclass/alternatives.eclass [hunk 1]
```diff
@@ -96,7 +94,7 @@ alternatives_makesym() {
 	# make sure it is in the prefix, allow it already to be in the prefix
 	SYMLINK=${EPREFIX}/${1#${EPREFIX}}
 	# this trick removes the trailing / from ${ROOT}
-	pref=${ROOT%/}
```
Comment (thesamesam) [inline eclass/alternatives.eclass:99]: Comment needs removing

File: eclass/java-osgi.eclass [hunk 2]
```diff
@@ -28,10 +28,7 @@ inherit java-utils-2
 # @ECLASS_VARIABLE: _OSGI_T
 # @INTERNAL
 # @DESCRIPTION:
-# We define _OSGI_T so that it does not contain a slash at the end.
-# According to Paludis guys, there is currently a proposal for EAPIs that
-# would require all variables to end with a slash.
-_OSGI_T="${T/%\//}"
+_OSGI_T="${T}"
 
 # must get Diego to commit something like this to portability.eclass
```
Comment (thesamesam) [inline eclass/java-osgi.eclass:33]: Convert into a professional TODO

File: eclass/postgres-multi.eclass [hunk 3]
```diff
@@ -177,3 +173,7 @@ postgres-multi_src_install() {
 postgres-multi_src_test() {
```
Comment (thesamesam) [inline eclass/postgres-multi.eclass:173]: Move above, order is wrong

File: eclass/selinux-policy-2.eclass [hunk 4]
```diff
@@ -219,7 +214,7 @@ selinux-policy-2_src_compile() {
 	for i in ${POLICY_TYPES}; do
 		# Support USE flags in builds
 		export M4PARAM="${makeuse}"
-		emake NAME=$i SHAREDIR="${ROOT%/}"/usr/share/selinux -C "${S}"/${i} || die "${i} compile failed"
+		emake NAME=$i SHAREDIR="${EPREFIX}"/usr/share/selinux -C "${S}"/${i} || die "${i} compile failed"
```
Comment (thesamesam) [inline eclass/selinux-policy-2.eclass:217]: Drop die

Comment (thesamesam) [reply eclass/selinux-policy-2.eclass:217]: Also iirc there's a bug to tag for this if you didn't already

File: eclass/dune.eclass [hunk 5]
```diff
@@ -8,41 +8,35 @@
...
-	6)
-		DEPEND="${RDEPEND} dev-ml/dune"
-		;;
-	*)
-		BDEPEND="dev-ml/dune dev-lang/ocaml"
-		DEPEND="${RDEPEND}"
-		;;
-esac
+DEPEND="${RDEPEND}"
+BDEPEND="dev-ml/dune dev-lang/ocaml"
```
Comment (a17r) [inline eclass/dune.eclass:39]: one line per dep?

File: eclass/aspell-dict-r1.eclass [hunk 6]
```diff
@@ -36,18 +36,15 @@ readonly ASPELL_SPELLANG=${PN/aspell-/}
...
-		;;
-	*)
-		die "${ECLASS}: EAPI ${EAPI:-0} not supported"
-		;;
+case ${EAPI} in
+	8) ;;
+	*) die "${ECLASS}: EAPI ${EAPI:-0} not supported" ;;
 esac
 
 EXPORT_FUNCTIONS src_configure src_install
```
Comment (a17r) [inline eclass/aspell-dict-r1.eclass:44]: do we move these below inherit or at end of file now?

File: eclass/linux-mod.eclass [hunk 7]
```diff
@@ -744,8 +736,8 @@ linux-mod_pkg_preinst() {
 	debug-print-function ${FUNCNAME} $*
 	[ -n "${MODULES_OPTIONAL_USE}" ] && use !${MODULES_OPTIONAL_USE} && return
 
-	[ -d "${D%/}/lib/modules" ] && UPDATE_DEPMOD=true || UPDATE_DEPMOD=false
-	[ -d "${D%/}/lib/modules" ] && UPDATE_MODULEDB=true || UPDATE_MODULEDB=false
+	[ -d "${D}/lib/modules" ] && UPDATE_DEPMOD=true || UPDATE_DEPMOD=false
+	[ -d "${D}/lib/modules" ] && UPDATE_MODULEDB=true || UPDATE_MODULEDB=false
```
Comment (a17r) [inline eclass/linux-mod.eclass:740]: double brackets?

File: eclass/usr-ldscript.eclass [hunk 8]
```diff
@@ -114,34 +114,34 @@ gen_usr_ldscript() {
 			# libdir=/lib because that messes up libtool files.
 			# Make sure we don't lose the specific version, so just modify the
 			# existing install_name
-			if [[ ! -w "${ED%/}/${libdir}/${tlib}" ]] ; then
-				chmod u+w "${ED%/}/${libdir}/${tlib}" # needed to write to it
+			if [[ ! -w "${ED}/${libdir}/${tlib}" ]] ; then
+				chmod u+w "${ED}/${libdir}/${tlib}" # needed to write to it
```
Comment (a17r) [inline eclass/usr-ldscript.eclass:118]: `|| die`?

File: eclass/fcaps.eclass [hunk 9]
```diff
@@ -41,8 +41,8 @@ IUSE="+filecaps"
 
 # Since it is needed in pkg_postinst() it must be in IDEPEND
 case ${EAPI} in
-	7) BDEPEND="filecaps? ( sys-libs/libcap )" ;& # fallthrough
-	6) RDEPEND="filecaps? ( sys-libs/libcap )" ;;
+	7) BDEPEND="filecaps? ( sys-libs/libcap )"
+	   RDEPEND="filecaps? ( sys-libs/libcap )" ;;
```
Comment (a17r) [inline eclass/fcaps.eclass:45]: slightly scruffy looking whitespace alignment here

File: eclass/apache-2.eclass [hunk 10]
```diff
@@ -388,7 +387,7 @@ setup_modules() {
 # This internal function generates the LoadModule lines for httpd.conf based on
 # the current module selection and MODULE_DEFINES
 generate_load_module() {
-	local def= endit=0 m= mod_lines= mod_dir="${ED%/}/usr/$(get_libdir)/apache2/modules"
+	local def= endit=0 m= mod_lines= mod_dir="${ED}/usr/$(get_libdir)/apache2/modules"
 
 	if use static; then
 		sed -i -e "/%%LOAD_MODULE%%/d" \
```
Comment (a17r) [inline eclass/apache-2.eclass:393]: this one has a missing `|| die` at the end

**Review by a17r — CHANGES_REQUESTED** [review]

**Review by a17r — APPROVED** [review]

File: eclass/apache-2.eclass [hunk 11]
```diff
@@ -3,8 +3,8 @@
 
 # @ECLASS: apache-2.eclass
 # @MAINTAINER:
-# polynomial-c@gentoo.org
```
Comment (mgorny) [inline eclass/apache-2.eclass:6]: I'm sorry for missing that during the retirement.

File: eclass/dune.eclass [hunk 12]
```diff
@@ -8,41 +8,39 @@
...
-		DEPEND="${RDEPEND} dev-ml/dune"
-		;;
-	*)
-		BDEPEND="dev-ml/dune dev-lang/ocaml"
-		DEPEND="${RDEPEND}"
-		;;
-esac
+RDEPEND="
+	>=dev-lang/ocaml-4:=[ocamlopt?]
+	dev-ml/dune:="
```
Comment (mgorny) [inline eclass/dune.eclass:39]: These days, I think it's better to do:

```suggestion
RDEPEND="
	>=dev-lang/ocaml-4:=[ocamlopt?]
	dev-ml/dune:=
"
```

File: eclass/ninja-utils.eclass [hunk 13]
```diff
@@ -76,11 +76,10 @@ get_NINJAOPTS() {
...
 # @DESCRIPTION:
 # Call Ninja, passing the NINJAOPTS (or converted MAKEOPTS), followed
-# by the supplied arguments. This function dies if ninja fails. Starting
-# with EAPI 6, it also supports being called via 'nonfatal'.
+# by the supplied arguments. This function dies if ninja fails.  It
+# also supports being called via 'nonfatal'.
 eninja() {
-	local nonfatal_args=()
-	[[ ${EAPI} != 5 ]] && nonfatal_args+=( -n )
+	local nonfatal_args=( -n )
```
Comment (mgorny) [inline eclass/ninja-utils.eclass:82]: Inline it.

**Review by mgorny — COMMENTED** [review]
No more comments from me. By the way, @arthurzam mentioned that one of the Python eclasses (single-r1?) is ready too, so maybe include that as well with the big cache regen.

**Review by thesamesam — APPROVED** [review]

**Review by arthurzam — APPROVED** [review]
I've done one scan over all changes in this PR, and all look fine to me.
Did multiple passes at python eclasses, and also look fine.
Thank you @SoapGentoo

**Review by ionenwks — APPROVED** [review]
Haven't had time to look all that in-depth but still went over the changes and looks fine to me.

Thanks for doing this.

**Review by ConiKost — APPROVED** [review]
Ack. ```aspell-dict-r1.eclass``` lgtm. Thank you for your work.

**Pull request closed without merging**